src/__test__/components/data-exploration/differential-expression-tool/DiffExprCompute.test.jsx

```diff
@@ -1,244 +1,286 @@
 
-const { Item } = Form;
+import DiffExprCompute from 'components/data-exploration/differential-expression-tool/DiffExprCompute';
+import { loadCellSets } from 'redux/actions/cellSets';
+import initialCellsetsState from 'redux/reducers/cellSets/initialState';
+import initialDiffExprState from 'redux/reducers/differentialExpression/initialState';
 
-const mockStore = configureMockStore([thunk]);
+import createTestComponentFactory from '__test__/test-utils/testComponentFactory';
 
-const initialState = {
-  cellSets: {
-    properties: {
-      'cluster-a': {
-        name: 'cluster a',
-        key: 'cluster-a',
-        cellIds: new Set([1, 2]),
-        color: '#00FF00',
-      },
-      'cluster-b': {
-        name: 'cluster b',
-        key: 'cluster-b',
-        cellIds: new Set([3, 4, 5]),
-        color: '#FF0000',
-      },
-      'cluster-c': {
-        name: 'cluster c',
-        key: 'cluster-c',
-        cellIds: new Set([6]),
-        color: '#0000FF',
-      },
-      'sample-a': {
-        name: 'sample a',
-        key: 'sample-a',
-        cellIds: new Set([1, 2]),
-        color: '#00FF00',
-      },
-      'sample-b': {
-        name: 'sample b',
-        key: 'sample-b',
-        cellIds: new Set([3, 4, 5]),
-        color: '#FF0000',
-      },
-      'sample-c': {
-        name: 'sample c',
-        key: 'sample-c',
-        cellIds: new Set([6]),
-        color: '#0000FF',
-      },
-      louvain: {
-        name: 'Louvain clusters',
-        key: 'louvain',
-        type: 'cellSets',
-        cellIds: new Set(),
-        rootNode: true,
-      },
-      scratchpad: {
-        name: 'Custom selections',
-        key: 'scratchpad',
-        type: 'cellSets',
-        cellIds: new Set(),
-        rootNode: true,
-      },
-      sample: {
-        name: 'Samples',
-        key: 'sample',
-        type: 'metadataCategorical',
-        cellIds: new Set(),
-        rootNode: true,
-      },
-    },
-    hierarchy: [
-      {
-        key: 'louvain',
-        children: [{ key: 'cluster-a' }, { key: 'cluster-b' }, { key: 'cluster-c' }],
-      },
-      {
-        key: 'sample',
-        children: [{ key: 'sample-a' }, { key: 'sample-b' }, { key: 'sample-c' }],
-      },
-      {
-        key: 'scratchpad',
-        children: [],
-      },
-    ],
-  },
-  differentialExpression: {
-    properties: {
-      data: [],
-      comparisonGroup: {},
-      loading: false,
-      error: false,
-      total: 0,
-    },
-    comparison: {
-      type: 'between',
-      group: {
-        between: {
-          cellSet: null,
-          compareWith: null,
-          basis: null,
-        },
-      },
-    },
-  },
+import mockAPI, { generateDefaultMockAPIResponses } from '__test__/test-utils/mockAPI';
+import fake from '__test__/test-utils/constants';
+
+enableFetchMocks();
+
+const createMockStore = configureMockStore([thunk]);
+
+const mockOnCompute = jest.fn();
+const experimentId = fake.EXPERIMENT_ID;
```

Comment:
nitpick: `experimentId` is used only 4 times in this code and it really is just `fake.EXPERIMENT_ID`. I don't see much value in having a variable for it

Reply:
I thought having this will just make it easier if we need to make changes, instead of having to remember to update multiple values. Aside from that, I also think we're more familar with using `experimentId` rather than having to remember to use `fake.EXPERIMENT_ID`,

Reply:
no problem, this is subjective, so up to you